eng/WpfArcadeSdk/tools/CodeGen/AvTrace/PresentationTraceSources.tt

```diff
@@ -0,0 +1,54 @@
+﻿<#@ template hostspecific="true" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ assembly name="$(NewtonsoftLocation)\Newtonsoft.Json.dll" #>
+<#@ assembly name="netstandard" #>
+<#@ assembly name="Microsoft.CSharp" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ import namespace="System.IO" #>
+<#@ import namespace="Newtonsoft.Json" #>
+<#@ output extension=".cs" #>
+<#@ parameter type="System.String" name="AvTraceMessageFiles" #>
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// NOTE: This file was generated by $(WpfCodeGenDir)AvTrace\PresentationTraceSources.tt.
+// Any manual updates to this file will overwritten.
+
+using MS.Internal;
+
+namespace System.Diagnostics
+{
+    /// <summary>Access point for TraceSources</summary>
+    public static partial class PresentationTraceSources
+    {
+<# 
+   foreach (var avTracMessageFile in AvTraceMessageFiles.Split(';'))
+   {
+		string fileName = this.Host.ResolvePath(avTracMessageFile);
+		string jsonText = File.ReadAllText(fileName);
+		dynamic jsonObj = JsonConvert.DeserializeObject(jsonText);
```

Comment:
If all you are doing is fetching the sources property and getting it's name, and do not need the `dynamic` functionality, then the `JsonDocument` (within System.Text.Json namespace) works for your use case here.

Reply:
Otherwise, if you need `dynamic`, then you would have to keep using `Newtonsoft.Json` since we don't support that functionality yet.

Reply:
i don't think there is a hard requirement, in fact i didn't like that i had used it, and i'm going to remove them entirely :)